decidim-core/lib/decidim/metric_registry.rb

```diff
@@ -52,12 +60,38 @@ def all
       metrics_manifests
     end
 
-    def highlighted
-      all.find_all { |manifest| manifest.highlighted == HIGHLIGHTED }.sort_by(&:position)
+    def filtered(highlight: nil, scope: nil, sort: nil)
+      result = all
+      unless highlight.nil?
+        result = if highlight
+                   highlighted(result)
+                 else
+                   not_highlighted(result)
+                 end
+      end
+      result = scoped(scope, result) unless scope.nil?
+      result = sorted(result) unless sort.nil?
```

Comment:
Same here, I'd prefer `if ... present` than `until .. nil` to avoid double negation.